lib/tcp/tcp_packet.ml

```diff
@@ -73,7 +73,6 @@ module Marshal = struct
   let unsafe_fill ~pseudoheader ~payload t buf options_len =
     let data_off = (sizeof_tcp / 4) + (options_len / 4) in
     let buf = Cstruct.sub buf 0 (data_off * 4) in
-    Cstruct.memset buf 0; (* autopad *)
```

Comment:
@balrajsingh , am I correct in presuming that this PR is the result of some erroneous behavior on the part of the TCP stack in some existing application?  If you point me at it, I can try putting a test together for you; it seems to me that you were motivated by noticing incorrect writes for options, so we could write a simple unit test for `Marshal.into_cstruct` which asserts that the returned buffer includes nonzero values representing the options passed.